### PR TITLE
Tsuru changes

### PR DESCRIPTION
**Pull request by fumieval**

### CODE REVIEW

File: src/Database/InfluxDB/JSON.hs [hunk 1]
```diff
@@ -79,17 +80,17 @@ parseResultsWithDecoder Decoder {..} row val0 = success <|> parseErrorObject val
...
-          values <- V.forM series $ \val -> do
-            (name, tags, columns, values) <- parseSeriesBody val
-            decodeFold $ V.forM values $ A.withArray "values" $ \fields -> do
-              assert (V.length columns == V.length fields) $ return ()
-              decodeEach $ row name tags columns fields
-          return $! join values
+
+      series <- fmap (join . V.map V.concat) $ V.forM results $ \val -> do
+        fmap pure (parseSeriesObject val)
+        <|> [] <$ parseStatementIdObject val
```
Comment (maoe) [inline src/Database/InfluxDB/JSON.hs]: This seems to be a bit complicated. I wonder if it's simpler if `parseSeriesObject` accepts empty objects.

Can you try something like this?

```diff
diff --git a/src/Database/InfluxDB/JSON.hs b/src/Database/InfluxDB/JSON.hs
index 4af9144..508dd44 100644
--- a/src/Database/InfluxDB/JSON.hs
+++ b/src/Database/InfluxDB/JSON.hs
@@ -81,10 +81,8 @@ parseResultsWithDecoder Decoder {..} row val0 = success <|> parseErrorObject val
     success = do
       results <- parseResultsObject val0

-      series <- fmap (join . V.map V.concat) $ V.forM results $ \val -> do
-        fmap pure (parseSeriesObject val)
-        <|> [] <$ parseStatementIdObject val
-        <|> parseErrorObject val
+      (join -> series) <- V.forM results $ \val ->
+        parseSeriesObject val <|> parseErrorObject val
       values <- V.forM series $ \val -> do
         (name, tags, columns, values) <- parseSeriesBody val
         decodeFold $ V.forM values $ A.withArray "values" $ \fields -> do
@@ -141,16 +139,12 @@ getTag tag tags = case HashMap.lookup tag tags of
   Just val -> return val

 parseResultsObject :: Value -> A.Parser (Vector A.Value)
-parseResultsObject = A.withObject "results" $ \obj -> obj .: "results"
+parseResultsObject = A.withObject "results" $ \obj -> do
+  fromMaybe V.empty <$> obj .:? "results"

 parseSeriesObject :: Value -> A.Parser (Vector A.Value)
 parseSeriesObject = A.withObject "series" $ \obj -> obj .: "series"

-parseStatementIdObject :: Value -> A.Parser Int
-parseStatementIdObject = withObject "results" $ \obj -> do
-  v <- obj .: "statement_id"
-  A.withScientific "statement_id" (return . round) v
-
 parseSeriesBody
   :: Value
   -> A.Parser (Maybe Text, HashMap Text Text, Vector Text, Array)
```

**Comment by fumieval** [comment]
I applied your suggestions. For `QueryResults ()`, I replaced it with `QueryResults Void`. I think this is  parametricistically cleaner.

```
instance QueryResults Void where
  parseResults _ = A.withObject "error" $ \obj ->
    maybe V.empty fail <$> obj .:? "error"
```

**Comment by maoe** [comment]
Hmm I thought I added that instance for some queries that returns an empty object. I'm skeptical `Void` is suitable for that. Let me remember what it was.

**Comment by maoe** [comment]
So it seems that `manage` doesn't fail even if there's an error:

On feature/influxdb-v1 branch:
```
λ> Manage.manage (queryParams "foo") "CRETE DATABAE boo"
*** Exception: BadRequest "error parsing query: found CRETE, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER
, SET, KILL at line 1, char 1" Request {
  host                 = "localhost"
  port                 = 8086
  secure               = False
  requestHeaders       = []
  path                 = "/query"
  queryString          = "?q=CRETE%20DATABAE%20boo"
  method               = "POST"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
```

On this branch:
```
λ> Manage.manage (queryParams "foo") "CRETE DATABAE boo"
it :: ()
```

**Comment by fumieval** [comment]
Fixed the bug, not sure what was wrong there.

The argument for Void is that `Vector Void` is isomorphic to `()`. Is there any use cases where you want to get `Vector ()` instead?

**Comment by maoe** [comment]
> Fixed the bug, not sure what was wrong there.

Looks good to me. Thanks.

> The argument for Void is that Vector Void is isomorphic to (). Is there any use cases where you want to get Vector () instead?

I now agree that `Void` is totally fine here.

Could you squash the fixup commits? I'll merge then then.

**Comment by fumieval** [comment]
OK, squashed

**Comment by maoe** [comment]
LGTM

**Pull request merged**